src/Microsoft.AspNetCore.Razor.Evolution/ITagHelperBoundAttributeDescriptorBuilder.cs

```diff
@@ -165,14 +165,11 @@ private IEnumerable<RazorDiagnostic> Validate()
 
             if (string.IsNullOrWhiteSpace(_name))
             {
-                if (_indexerNamePrefix == null)
```

Comment:
Talked offline. We definitely have tests for this guy so should investigate why those don't fail.